packages/mui-material/src/styles/createTheme.js

```diff
@@ -65,7 +65,7 @@ function createTheme(options = {}, ...args) {
     const traverse = (node, component) => {
       let key;
 
-      // eslint-disable-next-line guard-for-in, no-restricted-syntax
+      // eslint-disable-next-line guard-for-in
       for (key in node) {
```

Comment:
Not necessarily part of this PR, but is this `in` intentional? Is it required to traverse prototype properties here?

Another thing I noticed while sniffing around this code is that we can likely easily optimize occurences of `str.indexOf(search) === 0` to the more performant `string.startsWith(search)`

Reply:
Considering https://github.com/mui/material-ui/pull/13919/files#r241988655, this code may benefit from a more thorough review (cc @DiegoAndai) ;)
I'd like to keep this PR focused on eslint changes only, though.

Reply:
I'll verify first whether it's a real optimization